### DISCUSSION TITLE
Backward opc da compatibility

### DISCUSSION THREAD

**Discussion by AMo-Lhyfe** [FreeOpcUa/opcua-asyncio#377] [Q&A] [answered]
Hi, is there a built-in ability in asyncio to connect to opc da servers through the same library ?  Or do we need to implement a client that both can connect to DA and UA servers ? 
Thx

**Comment by oroulet** [comment, marked as answer]
there is no DA support in asyncua. if you want to support both DA and UA on server side, you need to implement two servers one for DA and one for UA. on client side you choose the library that matches your server
